systers_portal/meetup/views.py

```diff
@@ -1168,3 +1168,56 @@ def check_permissions(self, request):
         self.comment = get_object_or_404(Comment, pk=self.kwargs['comment_pk'])
         systersuser = get_object_or_404(SystersUser, user=request.user)
         return systersuser == self.comment.author
+
+
+class CancelMeetupLocationJoinRequestView(LoginRequiredMixin, SingleObjectMixin,
+                                          RedirectView):
+    """Cancel a join request to a meetup location view"""
+    model = MeetupLocation
+    permanent = False
+    raise_exception = True
+
+    # TODO: add `redirect_unauthenticated_users = True` when django-braces will
+    # reach version 1.5
+
+    def get_redirect_url(self, *args, **kwargs):
+        """Redirect to the page the user was previously on"""
+        return self.request.GET.get('current_url')
+
+    def get(self, request, *args, **kwargs):
+        """Attempt to cancel user join request towards a meetup location"""
+        user = get_object_or_404(User, username=self.kwargs.get('username'))
+        systersuser = get_object_or_404(SystersUser, user=user)
+
+        join_requests = self.get_object().join_requests.all()
+        members = self.get_object().members.all()
+
+        # for OK case
+        if systersuser not in members and systersuser in join_requests:
+            self.get_object().join_requests.remove(systersuser)
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Your request to join {0} meetup location was canceled.".format(self.get_object())
+            )
+        # for ALREADY_MEMBER case
```

Comment:
Instead of "# for ALREADY_MEMBER case" you can rewrite comments like in this method.  https://github.com/systers/portal/blob/3d870165ad0b93e55c084a2e463bede642ec6ff3/systers_portal/community/views.py#L227

Reply:
Made requested changes. Thanks for the review!